lib/elixir/lib/stream.ex

```diff
@@ -423,6 +423,10 @@ defmodule Stream do
   This operation will block the caller by the given interval
   every time a new item is streamed.
 
+  **Do not use this function** to generate number sequences. If
+  blocking the caller process is not necesary, use instead
+  `Stream.iterate(0, & &1 + 1)`.
```

Comment:
It is actually much more cheaper to use `Stream.interval(0)` than `Stream.iterate(0, & &1 + 1)`:
```elixir
iex> interval = Stream.interval(0)
iex> :timer.tc(Enum, :take, [interval, 100]) 
{36, ...}
iex> iterate = Stream.iterate(0, &(&1 + 1))
iex> :timer.tc(Enum, :take, [iterate, 100]) 
{184, ...}
```